src/gap-skipper.js

```diff
@@ -0,0 +1,86 @@
+import Ranges from './ranges';
+import videojs from 'video.js';
+
+export default class gapSkipper extends videojs.EventTarget {
+  constructor(options) {
+    super();
+    if (options.tech.options_.playerId) {
+      let self = this;
+
+      this.player = videojs(options.tech.options_.playerId);
+      this.tech_ = options.tech;
+      this.seeking = false;
+      this.consecutiveUpdates = 0;
+      this.timer = null;
+      this.playerState = null;
+      this.lastRecordedTime = null;
+
+      this.player.on('timeupdate', function() {
+        if (self.player.paused()) {
+          return;
+        }
+        let currentTime = self.player.currentTime();
+
+        if (self.consecutiveUpdates === 5 &&
+            currentTime === self.lastRecordedTime) {
+          // trigger waiting
+          if (self.playerState !== 'waiting') {
+            self.consecutiveUpdates = 0;
+            self.playerState = 'waiting';
+            self.skipTheGap();
+          }
+        } else if (currentTime === self.lastRecordedTime) {
+          self.consecutiveUpdates++;
+        } else {
+          self.consecutiveUpdates = 0;
+          self.lastRecordedTime = currentTime;
+        }
+      });
+
+      // Don't listen for waiting while seeking
+      this.player.on('seeking', function() {
+        self.seeking = true;
+      });
+
+      // Listen for waiting when finished seeking
+      this.player.on('seeked', function() {
+        self.seeking = false;
+      });
+
+      this.player.on('playing', function() {
+        self.player.on('waiting', self.skipTheGap);
+      });
+
+      this.player.on('error', function() {
+        if (self.timer) {
+          clearTimeout(self.timer);
+        }
+      });
+    }
+  }
+
+  skipTheGap() {
+    let self = this;
+
+    if (!this.seeking) {
```

Comment:
I would return from the function if we are seeking so we don't have to wrap so many this code in so many if statements. Example:
```javascript
if (this.seeking) {
  return
}